pkg/expect/expect.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package expect implements a small expect-style interface
+package expect
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/kr/pty"
+)
+
+type ExpectProcess struct {
+	cmd  *exec.Cmd
+	fpty *os.File
+	wg   sync.WaitGroup
+
+	cond  *sync.Cond // for broadcasting updates are avaiable
+	mu    sync.Mutex
+	lines []string
+	err   error
+}
+
+// NewExpect creates a new process for expect testing.
+func NewExpect(name string, arg ...string) (ep *ExpectProcess, err error) {
+	ep = &ExpectProcess{cmd: exec.Command(name, arg...)}
+	ep.cond = sync.NewCond(&ep.mu)
+	ep.cmd.Stderr = ep.cmd.Stdout
```

Comment:
How does this work? So overwriting stderr `io.Writer` combines
two writers into one, so that both stderr and stdout writes to same one
writer?